src/crawlee/crawlers/_basic/_basic_crawler.py

```
@@ -1088,6 +1108,10 @@ async def __run_task_function(self) -> None:
 
             self._statistics.record_request_processing_finish(statistics_id)
 
+        except RequestCollisionError as request_error:
+            context.request.no_retry = True
+            await self._handle_request_error(context, request_error)
+
```

So in case of a collision, the request will be aborted, right? Unless the user re-enqueues it in the `failed_request_handler`, which is kinda cumbersome. 

I imagine it can happen quite frequently that the session will get blocked and rotated out. I understand that silently using a new, different session might be confusing (but it might also be a viable option). Nevertheless, there should be an easier way to handle this in user code and it should be described in the documentation.

---

If we consider the case of the crawler working with authorization,

Each session is a separate authorized user.

Pages available for this user may not be available for another user. In this case, we cannot pass the request to another session. If a session becomes unavailable for some reason, then all we can do is give the user the opportunity to process these requests later.

Using this feature generally requires additional customizations to the `SessionPool`, so I don't think that having to use the `failed_request_handler` makes it any more cumbersome.

In general, I expect that users who will be using this know why they are doing it. 🙂

---

The tricky thing about `failed_request_handler` is that it is called after the request is marked as handled, so if you want to re-enqueue it with a different session, you need to make a new request. Are there any alternatives?

---

> Are there any alternatives?

Add a new handler for coliseums ?

> you need to make a new request.

That's right.

I would prefer it to be like this. A user created a `Request` with certain parameters, part of which is `session_id`. This `Request` has failed because one of the parameters is now invalid. I would expect that if the user wants to return it to the queue, a new `Request` with new parameters would be created